core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -621,6 +634,46 @@ private static AnnotatedValueResolver ofRequestObject(String name, AnnotatedElem
                 .build();
     }
 
+    private static AnnotatedValueResolver ofAttribute(String name,
+                                                      Attribute attr,
+                                                      AnnotatedElement annotatedElement,
+                                                      AnnotatedElement typeElement, Class<?> type,
+                                                      DescriptionInfo description) {
+
+        final List<AttributeKey<?>> attrKeys = new ArrayList<>();
```

Comment:
Could use `ImmutableList.builder()` to build an immutable list.

Reply:
Of course. i'm willing to replace `ArrayList` to `ImmutableList`! 
Thanks for your review 🙇‍♂️